core/psh/sysexec.c

```diff
@@ -25,6 +27,63 @@ void psh_sysexecinfo(void)
 }
 
 
+static int psh_sysexec_wldargmatch(char *wld, int argc, char **argv, char wldend, int argmaxlen)
+{
+	int len = 0, ii;
+	char *currc = wld, *nextc, *cmdend;
+
+	cmdend = strchr(wld, wldend);
+	currc = wld;
+	for (ii = 0; ii < argc; ii++) {
+
+		if ((nextc = strchr(currc, ' ')) == NULL || nextc > cmdend)
+			nextc = cmdend;
+		if ((nextc == cmdend) ^ (ii == (argc - 1))) /* exits if: last argc XOR last wldcrd argument */
+			return -1;
+
+		len = nextc - currc;
+		if (len > argmaxlen || len == 0 || strlen(argv[ii]) != len || memcmp(argv[ii], currc, len) != 0)
+			return -1;
+		if (*nextc != wldend && nextc != NULL)
+			currc = nextc + 1;
+	}
+	
+	return 1;
+}
+
+
+static int psh_sysexec_checkcommand(int argc, char **argv)
+{
+	int maxcdmlen = 80, acc = -1;
+	char wldbuff[maxcdmlen], *wldlist = PSH_WILDCARDS, *curr = wldlist, *nxtcomma;
+	FILE *wldfile;
+
+	/* check against /etc/wldcrds file */
+	if ((wldfile = fopen("/etc/wldcrds", "r")) != NULL) {
+		acc = 0;
+		while (fgets(wldbuff, maxcdmlen, wldfile) != NULL && acc != 1) {
+			if (!isprint(wldbuff[maxcdmlen - 2]) && psh_sysexec_wldargmatch(wldbuff, argc, argv, '\n', maxcdmlen-1) > 0)
+				acc = 1;
+			else
+				while (isprint(fgetc(wldfile))); /* skip to next wildcard */
```

Comment:
This code contains errors. You read every second line, if it is shorter than 80 chars.
If `!isprint(wldbuff[maxcdmlen - 2]` is true ( short line)
and `psh_sysexec_wldargmatch(wldbuff, argc, argv, '\n', maxcdmlen-1) > 0` is false,
The `while (isprint(fgetc(wldfile)));` loop reads next line, which is lost
What is the value of `!isprint(wldbuff[maxcdmlen - 2]` in the first iteration? `memset()` before `while`? 
Maybe use `getline()` instead of `fgets()`.

Reply:
Yes you are right. The logic was corrected. Stick to fgets, initialized `wldbuff[80] = ""` so this ensures zeroeness of buffer. Also